src/simplewallet/simplewallet.cpp

```diff
@@ -362,6 +362,27 @@ bool simple_wallet::seed_set_language(const std::vector<std::string> &args/* = s
   return true;
 }
 
+bool simple_wallet::change_password(const std::vector<std::string> &args)
+{ 
+  // prompts for a new password, this is not a new wallet so pass in false.
+  const auto pwd_container = tools::wallet2::password_prompt(false);
+  
+  try
+  {
+    m_wallet->rewrite(m_wallet_file, pwd_container->password());
+    m_wallet->store();
+  }
+  catch (int e)
```

Comment:
This isn't the correct exception type. I would specifically target:
```c++
catch (const wallet_logic_error& e)
{
    fail_msg_writer << tr("Error with wallet rewrite: ") << e.what();
}
```
A decent test to see if it works would be to change the wallet file ownership and permissions after the wallet loads. It should fail to write.